jormungandr/src/stats_counter.rs

```diff
@@ -1,6 +1,7 @@
+use chain_impl_mockchain::block::Block;
 use jormungandr_lib::time::SecondsSinceUnixEpoch;
 use std::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
```

Comment:
Wait, why are we adding synchronous locks again?
Everything that is used in async code should use Tokio locks for synchronization.

But since you are cloning this block to set it into that mutable context, a better solution might be as well to send it over an in-memory channel to the REST task, like we currently do for the explorer.

Reply:
Definetely should be a tokio RwLock.

@mzabaluev Let me check about the channel. Should we go for that @vincenthz ?

Reply:
I'm getting convinced that it shoudn't be a tokio lock on stuff that do:

```
lock
something that is : not blocking, relatively fast, not taking any other lock, not async
unlock
```

Reply:
@vincenthz Trouble is, the Tokio executor also relies on locks to implement a work-stealing scheduler, so any contentious locks in tasks will interfere with that.